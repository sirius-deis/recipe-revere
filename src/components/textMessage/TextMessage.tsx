import { FC } from "react";
import styles from './TextMessage.module.css'

const TextMessage: FC<{ message: string }> = ({ message }) => {
  return <p className={styles['text-message']}>{message}</p>
}

export default TextMessage