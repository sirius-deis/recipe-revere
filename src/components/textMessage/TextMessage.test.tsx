import { render, screen } from "@testing-library/react";
import TextMessage from "./TextMessage";

describe("TextMessage component", () => {
  it("should match snapshot", () => {
    const { container } = render(<TextMessage message="Hello, World!" />);
    expect(container).toMatchSnapshot();
  });

  it("should render the provided message", () => {
    render(<TextMessage message="Hello, World!" />);
    expect(screen.getByText("Hello, World!")).toBeInTheDocument();
  });

  it("should have a class name 'text-message'", () => {
    render(<TextMessage message="Hello, World!" />);
    expect(screen.getByText("Hello, World!")).toHaveClass("text-message");
  });
});