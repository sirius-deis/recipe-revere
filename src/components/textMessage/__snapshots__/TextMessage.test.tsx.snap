// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextMessage component should match snapshot 1`] = `
<div>
  <p
    class="text-message"
  >
    Hello, World!
  </p>
</div>
`;
